Methods.h

```diff
@@ -0,0 +1,50 @@
+#include "Classes.h"
+using namespace std;
+
+void Smartphone::add(float p, string n, float M, float A, float B)
+{
+    name = n;
+    price = p;
+    Memory_capacity = M;
+    Array_of_mobile_numbers = A;
+    Battery_capacity = B;
```

Comment:
You are using the wrong form of the variable name.

You should:
the_amount_of_permanent_memory_by_gb_ = permanent_memory;
array_of_mobile_numbers = model_name; // This change is not clear for what
battery_capacity_by_mah_ = battery_capacity;